File: android/delivery-api/GetTypes.java
// DocSection: delivery_api_get_types
// Tip: Find more about JavaRx SDK at https://docs.kontent.ai/androidandroid

// Prepares the DeliveryService configuration object
DeliveryOptions options = DeliveryOptions.builder()
        .projectId("<YOUR_PROJECT_ID>")
        .build();

// Initializes a DeliveryClient for Java projects
DeliveryClient client = new DeliveryClient(options);

// Gets 3 content types using a simple request
CompletionStage<ContentTypesListingResponse> types = client.getTypes(
    DeliveryParameterBuilder.params().page(null, 3).build()
);

// Gets 3 content types using RxJava
Observable.fromCompletionStage(types)
    .subscribe(new Observer<ContentTypesListingResponse>() {
        @Override
        public void onSubscribe(Disposable d) {
        }

        @Override
        public void onNext(ContentTypesListingResponse response) {
            // Gets content types from response
            List<ContentType> type = response.getTypes();
        }

        @Override
        public void onError(Throwable e) {
        }

        @Override
        public void onComplete() {
        }
    });
// EndDocSection

File: ios/delivery-api/GetTypes.swift
// DocSection: delivery_api_get_types
// Tip: Find more about Swift SDK at https://docs.kontent.ai/ios
import KenticoKontentDelivery
 
let client = DeliveryClient.init(projectId: "<YOUR_PROJECT_ID>")
 
client.getContentTypes(limit: 3, completionHandler: { (isSuccess, contentTypesResponse, error) in
    if !isSuccess {
        fail( "Response is not successful. Error: \(String(describing: error))" )
    }
     
    if let response = contentTypesResponse {
    // use content types here
    }
})
// EndDocSection